src/functions/edi/outbound/handler.ts

```diff
@@ -135,11 +135,26 @@ export const handler = async (
       })
     );
 
+    const deliveryResultsByStatus = deliveryResults.reduce(
+      (groupedResults: { "fulfilled": { value: DeliveryResult }[], "rejected": { reason: any }[]}, result) => {
+        const { status, ...rest } = result;
+        // @ts-ignore: TS doesn't correctly narrow `rest` as FulfilledResult or RejectedResult like it should
```

Comment:
`if (result.status === "fulfilled") {
          groupedResults[result.status].push(result)
        } else {
          groupedResults[result.status].push(result);
        }`
        
        Ross suggested this would solve your type issue

Reply:
tbh, I prefer the `@ts-ignore` comment over the `if/else` with identical logic for each branch